### PR TITLE
FLEX-5397 - Modify Meta Parent Compiler to Conditionally Include `hide_skip_approvals`

### PR DESCRIPTION
**Pull request by ramyadmz**
### Description

This pull request enhances the Meta Parent Policy Template Compiler to conditionally include the hide_skip_approvals field in the generated meta parent policy templates when it exists in the child policy templates.
It also removes prior manual modifications to meta parent policies : https://github.com/flexera-public/policy_templates/pull/2799

More context:[ Seeking Your Input: Proposed Solution for Conditional "Skip Approvals" Visibility](https://teams.microsoft.com/l/message/19:833373548e104af2a20b0216eda1ba7b@thread.skype/1728495063426?tenantId=91034d23-0b63-4943-b138-367d4dfac252&groupId=fb250818-e040-4a26-b207-61c3cd99fd6e&parentMessageId=1728495063426&teamName=Team%20Flexera%20One&channelName=Policy%20Support%20and%20Questions&createdTime=1728495063426)

### Issues Resolved

[FLEX-5397](https://flexera.atlassian.net/browse/FLEX-5397)

### Contribution Check List

- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -53,6 +53,7 @@ task :generate_policy_list do
       # 'deprecated' defaults to false unless explicitly set to true
       publish = !(publish == 'false' || publish == false)
       deprecated = deprecated == 'true' || deprecated == true
+      hide_skip_approvals = pp.parsed_info[:hide_skip_approvals]
```
Comment (XOmniverse) [inline Rakefile:56]: Can we confirm that this will work as expected for policy templates that do not have a hide_skip_approvals in their info() block?

**Review by XOmniverse — CHANGES_REQUESTED** [review]

File: Rakefile (hunk 1 above)
Comment (ramyadmz) [reply Rakefile:56]: Good point! Thanks for bringing it up.
I ran the Rakefile locally, and for policy templates without `hide_skip_approvals` in the info() block, the script adds it as nil in the JSON output. This aligns with how other optional fields such as `recommendation_type` are handled, so it should work as expected.
Screenshot 2024-11-08 104542

Comment (XOmniverse) [reply Rakefile:56]: For fields that toggle a setting, we want this to reflect what the policy is going to do. In those cases, it should show whatever the default functionality is if the value is not present; in this case, it should be false, similar to what you see with deprecated.

null is only used for non-boolean values that are non-applicable for some policy templates.

Comment (ramyadmz) [reply Rakefile]: got it, thanks. 
Applied at https://github.com/flexera-public/policy_templates/pull/2813/commits/755c47331075ce29b806cff633e2a1b89696c19d

File: Rakefile [hunk 2]
```diff
@@ -65,8 +69,7 @@ task :generate_policy_list do
       puts "Skipping #{pp.parsed_name} because publish flag set to a value other than 'true'"
     else
       # Get datetime for last time file was modified
-      commits = github_client.commits(repo_name, branch, path: file)
-      updated_at = commits.first.commit.author.date.utc.iso8601 if !commits.empty?
+      updated_at = `git log -1 --format="%cI" -- #{file}`.strip
```
Comment (XOmniverse) [inline Rakefile:72]: What's the reason for changing to a bash command instead of using the built-in github libraries?

**Review by XOmniverse — CHANGES_REQUESTED** [review]

File: Rakefile (hunk 2 above)
Comment (ramyadmz) [reply Rakefile]: My bad. I changed it for local testing. revert it: https://github.com/flexera-public/policy_templates/pull/2813/commits/7c7ccccea6e8486660a70d7b13622d116a84cbf4

**Review by XOmniverse — APPROVED** [review]
LGTM

**Pull request merged**